pkg/nsx/services/nsxserviceaccount/cluster.go

```diff
@@ -193,6 +156,94 @@ func (s *NSXServiceAccountService) CreateOrUpdateNSXServiceAccount(ctx context.C
 	return s.Client.Status().Update(ctx, obj)
 }
 
+// UpdateRealizedNSXServiceAccount checks if PI/CCP is created on NSXT for a realized NSXServiceAccount. If both PI/CCP
+// is missing, restore PI/CCP from realized NSXServiceAccount and Secret.
+func (s *NSXServiceAccountService) UpdateRealizedNSXServiceAccount(ctx context.Context, obj *v1alpha1.NSXServiceAccount) error {
+	normalizedClusterName := obj.Status.ClusterName
+
+	// check PI and CCP is missing
+	piObj := s.PrincipalIdentityStore.GetByKey(normalizedClusterName)
+	ccpObj := s.ClusterControlPlaneStore.GetByKey(normalizedClusterName)
+	if piObj != nil && ccpObj != nil {
+		return nil
+	} else if (piObj != nil) != (ccpObj != nil) {
+		return fmt.Errorf("PI/CCP doesn't match")
+	}
+	_, err := s.NSXClient.ClusterControlPlanesClient.Get(siteId, enforcementpointId, normalizedClusterName)
+	if err == nil {
+		return fmt.Errorf("CCP store is not synchronized")
+	}
+	switch err.(type) {
+	case vapierrors.NotFound:
+	default:
+		return err
+	}
+
+	// check Secret
+	log.Info("Start to restore realized", "nsxserviceaccount", types.NamespacedName{Name: obj.Name, Namespace: obj.Namespace})
```

Comment:
may I know the reason to put "nsxserviceaccount" in a single phrase other than putting it togerther with "Start to restore realized"?

Reply:
This `log.Info` makes same behavior as `klog.InfoS`, thus is means `"Start to restore realized, nsxserviceaccount=%v", namespacednamme`

Reply:
I feel that this line can be better written as below.

```
log.Info("Start to restore realized Kubernetes resource", "nsxserviceaccount", ...
```